### PR TITLE
Configured Flow support and added type information.  Enabled linting.

### PR DESCRIPTION
**Pull request by the-t-in-rtf**
I finished my initial work to add Flow to my sample project.  In addition to general feedback, I could use help understanding the remaining warnings.

cc: @simonbates @chosww

### CODE REVIEW

File: src/MultiSelect.js [hunk 1]
```diff
@@ -1,51 +1,63 @@
...
-export default class MultiSelect extends React.Component {
-    constructor(props) {
+type OptionDef = {
+    value: string,
+    label?: string
+}
+
+type MultiSelectProps = {
+    title: string,
+    selectedValues?: Array<Object>,
```
Comment (chosww) [inline src/MultiSelect.js]: 'Object' is an alias to 'any' and will be deprecated and removed in a future version of Flow -- https://flow.org/en/docs/types/objects/ at the bottom of the page

File: src/MultiSelect.js [hunk 2]
```diff
@@ -1,51 +1,63 @@
...
-        let selectedValues = Object.keys(selectedValueMap);
+        const selectedValues: Array<string> = Object.keys(selectedValueMap);
         this.setState({
             selectedValues: selectedValues
         });
 
+        // TODO: Discuss why type coverage is not picked up based on the annotations above.
         if (this.props.stateChangeCallbacks) {
-            this.props.stateChangeCallbacks.forEach((callback) => {
+            this.props.stateChangeCallbacks.forEach((callback: Function) => {
```
Comment (chosww) [inline src/MultiSelect.js]: In the coding environment, we specify function type as an anonymous arrow function like 'callback: () => void'; you can also specify input parameter type to the function like, 'callback: (index: number) => void'.

Comment (the-t-in-rtf) [reply src/MultiSelect.js]: Thanks, that fixes the last coverage warning.

File: src/MultiSelect.js (hunk 1 above)
Comment (the-t-in-rtf) [reply src/MultiSelect.js]: Good to know, I am opposed in general to using `any`, `Object`, or `Function` if we can help it.  I removed nearly all of them on the latest commit (still reading up on refs).

**Comment by the-t-in-rtf** [comment]
There are a few small things I haven't managed yet, but my type coverage is on par with the main codebase.  I'd like to understand how to properly type references, but maybe the way I'm using them is sub-optimal anyway.  Happy to merge this and move on and revisit if it comes up working on the main codebase.

**Pull request merged**